### PR TITLE
Added documentation for collector GRPCv2 migration

### PR DESCRIPTION
**Pull request by rasmus-kirk**
## Changes

Added documentation for the Collector GRPCv2 migration and fixed a few minor things.

### CODE REVIEW

**Comment by rasmus-kirk** [comment]
The docker compose configurations needs to be updated. I'll do it as soon as https://github.com/Concordium/concordium-node/issues/793 gets merged

**Comment by abizjak** [comment]
@dg-concordium The context for this is the next node release. This should only be merged then. It might have to be split into multiple PRs for separate testnet/mainnet releases.

**Comment by dg-concordium** [comment]
> @dg-concordium The context for this is the next node release. This should only be merged then. It might have to be split into multiple PRs for separate testnet/mainnet releases.

Yes, I will split it. When I look at these topics, I think adding a heading for node collector might be a good idea. The information gets lost, particularly in the Ubuntu topics.

**Comment by abizjak** [comment]
> > @dg-concordium The context for this is the next node release. This should only be merged then. It might have to be split into multiple PRs for separate testnet/mainnet releases.
> 
> Yes, I will split it. When I look at these topics, I think adding a heading for node collector might be a good idea. The information gets lost, particularly in the Ubuntu topics.

Sounds like a good idea.

File: source/mainnet/net/guides/run-node.rst [hunk 1]
```diff
@@ -378,28 +419,26 @@ To retrieve mainnet node logs run:
...
        - "10000:10000"
        - "20000:20000"
+
        volumes:
        # The node's database should be stored in a persistent volume so that it
        # survives container restart. In this case we map the **host** directory
-       # /var/lib/concordium-mainnet to be used as the node's database directory.
-       - /var/lib/concordium-mainnet:/mnt/data
+       # ~/tmp/mainnet to be used as the node's database directory.
+       - /tmp/mainnet:/mnt/data
```
Comment (abizjak) [inline source/mainnet/net/guides/run-node.rst:440]: I don't think this change should have been made. It is not related to any code changes we have done, and what was there was fine.

File: source/mainnet/net/guides/run-node.rst [hunk 2]
```diff
@@ -378,28 +419,26 @@ To retrieve mainnet node logs run:
...
        # Exposed ports. The ports the node listens on inside the container (defined
        # by `CONCORDIUM_NODE_LISTEN_PORT` and `CONCORDIUM_NODE_RPC_SERVER_PORT`)
        # should match what is defined here. When running multiple nodes the
        # external ports should be changed so as not to conflict.
-       # In the mapping below, the first port is the `host` port, and the second
-       # port is the `container` port. When the `container` port is changed the
-       # relevant environment variable listed above must be changed as well. For
-       # example, changing `10000:10000` to `10000:13000` would mean that
-       # `CONCORDIUM_NODE_RPC_SERVER_PORT` should be set to `13000`. Otherwise
-       # the node's gRPC interface will not be available from the host.
```
Comment (abizjak) [inline source/mainnet/net/guides/run-node.rst:393]: Please keep this explanation in. It was specifically requested.

File: source/mainnet/net/guides/run-node.rst [hunk 3]
```diff
@@ -336,18 +377,18 @@ To retrieve mainnet node logs run:
        environment:
          # Environment specific configuration
          # The url where IPs of the bootstrap nodes can be found.
-         - CONCORDIUM_NODE_CONNECTION_BOOTSTRAP_NODES=bootstrap.mainnet.concordium.software:8888
+         - CONCORDIUM_NODE_CONNECTION_BOOTSTRAP_NODES=bootstrap.mainnet.concordium.com:8888
          # Where the genesis is located
          - CONCORDIUM_NODE_CONSENSUS_GENESIS_DATA_FILE=/mainnet-genesis.dat
-         # The url of the catchup file. This speeds up the catchup process.
-         - CONCORDIUM_NODE_CONSENSUS_DOWNLOAD_BLOCKS_FROM=https://catchup.mainnet.concordium.software/blocks.idx
```
Comment (abizjak) [inline source/mainnet/net/guides/run-node.rst:343]: Keep this in.

File: source/mainnet/net/guides/run-node.rst [hunk 4]
```diff
@@ -119,28 +158,26 @@ To run a node on testnet use the following configuration file and follow the ste
...
        - "10001:10001"
        - "20001:20001"
+
        volumes:
        # The node's database should be stored in a persistent volume so that it
        # survives container restart. In this case we map the **host** directory
-       # /var/lib/concordium-testnet to be used as the node's database directory.
-       - /var/lib/concordium-testnet:/mnt/data
+       # ~/tmp/testnet to be used as the node's database directory.
+       - /tmp/testnet:/mnt/data
```
Comment (abizjak) [inline source/mainnet/net/guides/run-node.rst:179]: Please keep it as it was. This is unrelated to any changes with the collector.

File: source/mainnet/net/guides/run-node.rst [hunk 5]
```diff
@@ -80,15 +119,15 @@ To run a node on testnet use the following configuration file and follow the ste
          - CONCORDIUM_NODE_CONNECTION_BOOTSTRAP_NODES=bootstrap.testnet.concordium.com:8888
          # Where the genesis is located
          - CONCORDIUM_NODE_CONSENSUS_GENESIS_DATA_FILE=/testnet-genesis.dat
-         # The url of the catchup file. This speeds up the catchup process.
-         - CONCORDIUM_NODE_CONSENSUS_DOWNLOAD_BLOCKS_FROM=https://catchup.testnet.concordium.com/blocks.idx
```
Comment (abizjak) [inline source/mainnet/net/guides/run-node.rst:84]: Keep this in. No reason to remove it.

File: source/mainnet/net/guides/run-node.rst [hunk 6]
```diff
@@ -80,15 +119,15 @@ To run a node on testnet use the following configuration file and follow the ste
...
+
          # General node configuration Data and config directories (it's OK if they
          # are the same). This should match the volume mount below. If the location
          # of the mount inside the container is changed, then these should be
          # changed accordingly as well.
          - CONCORDIUM_NODE_DATA_DIR=/mnt/data
          - CONCORDIUM_NODE_CONFIG_DIR=/mnt/data
-         # The port on which the node will listen for incoming connections. This is a
+
+         # port on which the node will listen for incoming connections. This is a
```
Comment (abizjak) [inline source/mainnet/net/guides/run-node.rst:130]: I think we should keep the capitalization as it was. It was consistent to start all these sentences with a capital letter.

File: source/mainnet/net/guides/run-node.rst [hunk 7]
```diff
@@ -151,16 +188,19 @@ To run a node on testnet use the following configuration file and follow the ste
...
+
          # Collection settings.
          # How often to collect the statistics from the node.
          - CONCORDIUM_NODE_COLLECTOR_COLLECT_INTERVAL=5000
          # The URL where the node can be reached. Note that this will use the
          # docker created network which maps `testnet-node` to the internal IP of
          # the `testnet-node`. If the name of the node service is changed from
          # `testnet-node` then the name here must also be changed.
-         - CONCORDIUM_NODE_COLLECTOR_GRPC_HOST=http://testnet-node:10001
+         - CONCORDIUM_NODE_COLLECTOR_GRPC_HOST=http://testnet-node:20001
```
Comment (abizjak) [inline source/mainnet/net/guides/run-node.rst:202]: This is really the only change that needed to be made. The rest is unrelated.

File: source/mainnet/net/guides/run-node.rst [hunk 8]
```diff
@@ -410,16 +449,19 @@ To retrieve mainnet node logs run:
        environment:
          # Settings that should be customized by the user.
          - CONCORDIUM_NODE_COLLECTOR_NODE_NAME=docker-test-mainnet
+
          # Environment specific settings.
-         - CONCORDIUM_NODE_COLLECTOR_URL=https://dashboard.mainnet.concordium.software/nodes/post
+         - CONCORDIUM_NODE_COLLECTOR_URL=https://dashboard.mainnet.concordium.com/nodes/post
```
Comment (abizjak) [inline source/mainnet/net/guides/run-node.rst:454]: `mainnet` is `.software`. Please revert.

File: source/mainnet/net/guides/run-node.rst [hunk 9]
```diff
@@ -410,16 +449,19 @@ To retrieve mainnet node logs run:
...
+
          # Collection settings.
          # How often to collect the statistics from the node.
          - CONCORDIUM_NODE_COLLECTOR_COLLECT_INTERVAL=5000
          # The URL where the node can be reached. Note that this will use the
          # docker created network which maps `mainnet-node` to the internal IP of
          # the `mainnet-node`. If the name of the node service is changed from
          # `mainnet-node` then the name here must also be changed.
-         - CONCORDIUM_NODE_COLLECTOR_GRPC_HOST=http://mainnet-node:10000
+         - CONCORDIUM_NODE_COLLECTOR_GRPC_HOST=http://mainnet-node:20000
```
Comment (abizjak) [inline source/mainnet/net/guides/run-node.rst:463]: This is the only change that needed to be made in this configuration.

File: source/mainnet/net/guides/run-node-macos.rst [hunk 10]
```diff
@@ -258,6 +259,31 @@ If you want to change whether the node services start automatically, you have tw
 
          $sudo rm "/Library/LaunchDaemons/software.concordium.testnet.node-collector.plist"
 
+.. _node-collector-configuration-macos:
+
+Node collector configuration
```
Comment (abizjak) [inline source/mainnet/net/guides/run-node-macos.rst:264]: We should make sure to note this migration in the release notes @dg-concordium

**Comment by rasmus-kirk** [comment]
Once this passes review, I will update [this PR](https://github.com/Concordium/concordium-node/pull/811) with the example configs approved here.

File: source/mainnet/net/guides/run-node-macos.rst (hunk 10 above)
Comment (dg-concordium) [reply source/mainnet/net/guides/run-node-macos.rst:264]: I've added links in the release note per platform in the testnet release PR.

File: source/mainnet/net/guides/run-node-macos.rst [hunk 11]
```diff
@@ -258,6 +259,31 @@ If you want to change whether the node services start automatically, you have tw
...
+============================
+
+Since version 5.3.0 of the node, the collector uses the GRPC V2 interface. Therefore, in order to run the collector, it is required that the node which the collector connects to has the GRPC V2 interface enabled.
```
Comment (abizjak) [inline source/mainnet/net/guides/run-node-macos.rst:267]: This should ideally be 5.3.2, or perhaps we can just say `5.3.*` to avoid updating it.

File: source/mainnet/net/guides/run-node-windows.rst [hunk 12]
```diff
@@ -251,6 +251,11 @@ You can improve the performance by downloading the blocks from an out-of-band ca
 
 #. Go to the relevant dashboard to monitor when the node has caught up with its peers on the blockchain. To see this, compare the finalized length of the chain with the length of your node. If they match, your node has caught up.
 
+Node collector configuration
+============================
+
+Since version 5.3.0 of the node, the collector uses the GRPC V2 interface. Therefore, in order to run the collector, it is required that the node which the collector connects to has the GRPC V2 interface enabled. For information about how to do this, see :ref:`the note at the top of this topic<install-upgrade-node-windows>`.
```
Comment (abizjak) [inline source/mainnet/net/guides/run-node-windows.rst:257]: Same here.

File: source/mainnet/net/nodes/run-node-ubuntu.rst [hunk 13]
```diff
@@ -93,18 +93,38 @@ If the node is well behind the head of the chain, you can speed up the startup b
...
       $sudo systemctl start concordium-testnet-node.service
 
 After the node is caught up remove the out of band catchup configuration to speed up further node restarts.
 
+.. _node-collector-ubuntu-testnet:
+
+Node collector configuration
+============================
+
+Since version 5.3.0 of the node, the collector uses the GRPC V2 interface. Therefore, in order to run the collector, it is required that the node which the collector connects to has the GRPC V2 interface enabled.
```
Comment (abizjak) [inline source/mainnet/net/nodes/run-node-ubuntu.rst:113]: Same version change should be done here.

**Review by abizjak — APPROVED** [review]
Looks OK assuming the version is updated as requested.

**Review by mh-concordium — APPROVED** [review]
Looks good.
